framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
Usage of ternary operator introduces a "magic string" - `UTF-8`.
Keeping default encoding at Application field is a one thing, but put putting same string constant all over the code is a different one.
While in current state `UTF-8` is covering most of the needs and is de-facto standard it may change in the future. After several years we may use `UTF-16` for example.

We should either introduce a static level for default encoding (e.g. `Yii::$encoding` or `Yii::encoding()` or `Yii::$defaultEncoding`) or abandon the whole idea.